timdex_dataset_api/dataset.py

```diff
@@ -108,32 +107,30 @@ class TIMDEXDataset:
 
     def __init__(
         self,
-        location: str | list[str],
+        location: str,
         config: TIMDEXDatasetConfig | None = None,
     ):
         """Initialize TIMDEXDataset object.
 
         Args:
-            location (str | list[str]): Local filesystem path or an S3 URI to
-                a parquet dataset. For partitioned datasets, set to the base directory.
+            location (str ): Local filesystem path or an S3 URI to a parquet dataset.
```

Comment:
This is a big change, that reverberates throughout.  By limiting to only `str` here, we're forced (get to!) remove the `list[str]` options anywhere `self.location` was used.